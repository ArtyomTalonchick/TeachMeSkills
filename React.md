База:
- [Создание React-приложения](https://ru.reactjs.org/docs/create-a-new-react-app.html)
- [CDN](https://ru.reactjs.org/docs/cdn-links.html)
- [JSX](https://ru.reactjs.org/docs/introducing-jsx.html)
- [Компоненты и пропсы](https://ru.reactjs.org/docs/components-and-props.html)
  - [Функциональные и классовые компоненты](https://ru.reactjs.org/docs/components-and-props.html#function-and-class-components)
  - [Пропсы можно только читать](https://ru.reactjs.org/docs/components-and-props.html#props-are-read-only)

Состояние и жизненный цикл:
- [Состояние и жизненный цикл](https://ru.reactjs.org/docs/react-component.html#the-component-lifecycle)
- `this.state` и `this.setState`
- Методы жизненного цикла
- [Правила использования состояния](https://ru.reactjs.org/docs/state-and-lifecycle.html#using-state-correctly)