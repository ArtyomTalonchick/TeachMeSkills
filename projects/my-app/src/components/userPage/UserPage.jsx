import { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { getUser } from '../../api/usersApi';

const UserPage = (props) => {

    const [user, setUser] = useState({});

    const params = useParams();

    useEffect(() => {
        getUser(params.login)
            .then((response) => setUser(response.data));
    }, []);

    useEffect(() => {
        console.log(user);
    });

    return (
        <div>
            {user.login}
        </div>
    )
}



export default UserPage;