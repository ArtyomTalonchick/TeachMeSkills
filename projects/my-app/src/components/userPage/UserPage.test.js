import React from 'react';
import { render } from "@testing-library/react"; 
import "@testing-library/jest-dom/extend-expect";

import axios from "axios";

import UserPage from './UserPage';

jest.mock("axios");
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: () => ({
        login: '1',
    }),
  }));

const response = {
    data: {
        "id": 1,
        "avatar_url": "https://avatars.githubusercontent.com/u/1?v=4",
        "login": "AlexName",
        "html_url": "https://github.com/mojombo",
        "type": "User",
    }
}

test('match snapshot', async () => {
    axios.get.mockResolvedValueOnce(response);

    const { asFragment, findByText } = render(<UserPage />)

    
    expect(await findByText("AlexName")).toBeInTheDocument();

        
    expect(asFragment()).toMatchSnapshot();
});