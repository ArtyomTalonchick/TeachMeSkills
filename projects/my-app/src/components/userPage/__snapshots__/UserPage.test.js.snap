// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<DocumentFragment>
  <div>
    AlexName
  </div>
</DocumentFragment>
`;
